proxy/http/HttpProxyAPIEnums.h

```diff
@@ -29,18 +29,27 @@
 
 #pragma once
 
-// This is use to signal apidefs.h to not define these again.
-#ifndef _HTTP_PROXY_API_ENUMS_H_
-#define _HTTP_PROXY_API_ENUMS_H_
-
 /// Server session sharing values - match
 typedef enum {
+  TS_SERVER_SESSION_SHARING_MATCH_IP,
+  TS_SERVER_SESSION_SHARING_MATCH_HOSTONLY,
+  TS_SERVER_SESSION_SHARING_MATCH_HOSTSNISYNC,
+  TS_SERVER_SESSION_SHARING_MATCH_SNI,
+  TS_SERVER_SESSION_SHARING_MATCH_CERT,
   TS_SERVER_SESSION_SHARING_MATCH_NONE,
   TS_SERVER_SESSION_SHARING_MATCH_BOTH,
-  TS_SERVER_SESSION_SHARING_MATCH_IP,
-  TS_SERVER_SESSION_SHARING_MATCH_HOST
+  TS_SERVER_SESSION_SHARING_MATCH_HOST,
 } TSServerSessionSharingMatchType;
 
+typedef enum {
+  TS_SERVER_SESSION_SHARING_MATCH_MASK_NONE        = 0,
+  TS_SERVER_SESSION_SHARING_MATCH_MASK_IP          = 0x1,
+  TS_SERVER_SESSION_SHARING_MATCH_MASK_HOSTONLY    = 0x2,
+  TS_SERVER_SESSION_SHARING_MATCH_MASK_HOSTSNISYNC = 0x4,
+  TS_SERVER_SESSION_SHARING_MATCH_MASK_SNI         = 0x8,
+  TS_SERVER_SESSION_SHARING_MATCH_MASK_CERT        = 0x10
+} TSServerSessionSharingMatchMask;
```

Comment:
Instead of this maybe:
```
inline unsigned serverSessionSharingMatchMask(TSServerSessionSharingMatchType t)
{
  ink_assert(t < TS_SERVER_SESSION_SHARING_MATCH_NONE);
  return 1 << t;
}
```